src/app/query/bitcoin/ordinals/inscriptions/inscriptions.query.ts

```diff
@@ -0,0 +1,57 @@
+import { useCallback, useMemo } from 'react';
+
+import { useQueries } from '@tanstack/react-query';
+
+import {
+  combineInscriptionResults,
+  createInscriptionByXpubQuery,
+  createNumberOfInscriptionsFn,
+  filterUninscribedUtxosToRecoverFromTaproot,
+  useBitcoinClient,
+  useGetTaprootUtxosByAddressQuery,
+  utxosToBalance,
+} from '@leather.io/query';
+
+import { useCurrentAccountIndex } from '@app/store/accounts/account';
+import { useCurrentBitcoinAccountXpubs } from '@app/store/accounts/blockchain/bitcoin/bitcoin.hooks';
+import { useCurrentTaprootAccount } from '@app/store/accounts/blockchain/bitcoin/taproot-account.hooks';
+
+interface UseInscriptionArgs {
+  xpubs: string[];
+}
+export function useInscriptions({ xpubs }: UseInscriptionArgs) {
+  const client = useBitcoinClient();
+  const queries = xpubs.map(xpub => createInscriptionByXpubQuery(client, xpub));
+  return useQueries({ queries, combine: combineInscriptionResults });
+}
+
+export function useNumberOfInscriptionsOnUtxo() {
+  const xpubs = useCurrentBitcoinAccountXpubs();
+  const inscriptionQueries = useInscriptions({ xpubs });
+
+  // Unsafe as implementation doesn't wait for all results to be successful,
```

Comment:
Can we improve this to wait or is it OK for now? 

Maybe `inscriptions.query` should move to the query package?

Reply:
As much logic as possible is in query package, I think, avoiding wrappers of `useQuery` themselves. Even reused logic to combine the queries. "Merging" at client level gives us flexibility to use them in other ways.

Added this comment because yeah the code just seems very unsafe. If the query is loading, assume there are none 🤷🏼‍♂️